src/changeTimeTrackingRequests/changeTimeTrackingRequest.test.ts

```diff
@@ -0,0 +1,168 @@
+
+import { ErrorReason, TimeTacApiError } from '../errors';
+import { RequestParamsBuilder } from '../utils/params/requestParams';
+import { ReadRawResponse } from '../utils/response/readRawResponse';
+import { ChangeTimeTrackingRequestEndpoint } from './index';
+import { ChangeTimeTrackingRequest } from './types';
+
+type Resource = ChangeTimeTrackingRequest;
+const endpoint: ChangeTimeTrackingRequestEndpoint = new ChangeTimeTrackingRequestEndpoint({ account: 'testingAccount' });
+
+describe('changeTimeTrackingRequest.read', () => {
+  const readPath = `${endpoint.getResourcePath()}/read`;
+
+  test('with status 200 and Success true and single result', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+
+    mock.onGet(readPath).reply(200, { Success: true, NumResults: 1, Results: [record] });
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual).toStrictEqual([record]);
+  });
+
+  test('with status 200 and Success true and more results', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+
+    mock.onGet(readPath).reply(200, { Success: true, NumResults: 5, Results: [record, record, record, record, record] });
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual).toStrictEqual([record, record, record, record, record]);
+  });
+
+  test('with status 200 and Success true and params', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+    const body = { params: { id: '42', _op__id: 'eq' } };
+    const requestParams = new RequestParamsBuilder<Resource>().eq('id', 42).build();
+
+    mock.onGet(readPath, body).reply(200, { Success: true, NumResults: 5, Results: [record] });
+    const actual: Promise<Resource[]> = endpoint.read(requestParams);
+    expect(await actual).toStrictEqual([record]);
+  });
+
+  test('with status 200 and Success false', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+    const apiResponse = { Success: false, NumResults: 1, Results: [record] };
+
+    mock.onGet(readPath).reply(200, apiResponse);
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual.catch((err: typeof apiResponse) => err)).toStrictEqual(apiResponse);
+  });
+
+  test('with status 400 and Success false', async () => {
+    const mock = new AxiosMockAdapter(axios);
+    const record = createMock<Resource>();
+    const apiResponse = { Success: false, NumResults: 1, Results: [record] };
+
+    mock.onGet(readPath).reply(400, apiResponse);
+    const actual: Promise<Resource[]> = endpoint.read();
+    expect(await actual.catch((err: { message: string }) => err.message)).toMatch('Request failed with status code 400');
+  });
+});
+
+describe('changeTimeTrackingRequestRead.readRaw', () => {
+  const readPath = `${endpoint.getResourcePath()}/read`;
```

Comment:
@tomi-timetac I think they are differenct, read vs. readRaw, I am not sure. @whosshell could you comment if both tests are needed?

Reply:
I would leave it for now. To have more clear answer I would have to check what is `epicStrategyBuilder` doing.
If more information is needed contact me directly.